test/index/testing_index_util.cpp

```diff
@@ -36,8 +36,11 @@ void TestingIndexUtil::BasicTest(const IndexType index_type) {
   std::vector<ItemPointer *> location_ptrs;
 
   // INDEX
-  std::unique_ptr<index::Index> index(
-      TestingIndexUtil::BuildIndex(index_type, false));
+  std::unique_ptr<index::Index, std::function<void(index::Index *)>> index(
+      TestingIndexUtil::BuildIndex(index_type, false),
+      [](index::Index *index) {
```

Comment:
If I put this in a separate function and reuse it, the binary size(in the release mode) gets bigger by 169 bytes.  This is because the binary needs some symbol information for handling a separate function.  However, I would buy your argument because it is less error-prone.